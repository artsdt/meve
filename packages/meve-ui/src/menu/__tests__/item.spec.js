import Menu from '..'
import MenuItem from '../../menu-item'
import { mount } from '@vue/test-utils'

test('test menu item props', async () => {
  const component = {
    components: {
      [Menu.name]: Menu,
      [MenuItem.name]: MenuItem,
    },
    template: `
      <m-menu>
        <m-menu-item name="1" label="level 1" icon="error" />
      </m-menu>
    `,
  }

  const wrapper = mount(component)
  expect(wrapper.html()).toMatchSnapshot()
  wrapper.destroy()
})

test('test menu item disabled', async () => {
  const handleItemChange = jest.fn()

  const component = {
    components: {
      [Menu.name]: Menu,
      [MenuItem.name]: MenuItem,
    },
    methods: {
      handleItemChange,
    },
    data: () => ({
      disabled: true,
    }),
    template: `
      <m-menu @update:selected-names="handleItemChange">
        <m-menu-item name="1" label="level 1" icon="error" :disabled="disabled" />
      </m-menu>
    `,
  }

  const wrapper = mount(component)
  expect(wrapper.html()).toMatchSnapshot()

  await wrapper.find('.m-menu-item').trigger('click')
  expect(handleItemChange).toHaveBeenCalledTimes(0)

  await wrapper.setData({ disabled: false })
  await wrapper.find('.m-menu-item').trigger('click')
  expect(handleItemChange).toHaveBeenCalledTimes(1)

  wrapper.destroy()
})

test('test menu item pass render function', async () => {
  const component = {
    components: {
      [Menu.name]: Menu,
      [MenuItem.name]: MenuItem,
    },
    methods: {
      renderIcon(h) {
        return h('div', 'custom icon')
      },

      renderLabel(h) {
        return h('div', 'custom label')
      },
    },
    template: `
      <m-menu>
        <m-menu-item name="1" :label="renderLabel" :icon="renderIcon" />
      </m-menu>
    `,
  }

  const wrapper = mount(component)
  expect(wrapper.html()).toMatchSnapshot()

  wrapper.destroy()
})

test('test menu item icon pass object', async () => {
  const component = {
    components: {
      [Menu.name]: Menu,
      [MenuItem.name]: MenuItem,
    },
    template: `
      <m-menu>
        <m-menu-item name="1" label="level i" :icon="{ name: 'error', color: 'red', size: 30 }" />
      </m-menu>
    `,
  }

  const wrapper = mount(component)
  expect(wrapper.html()).toMatchSnapshot()

  wrapper.destroy()
})

test('test menu item slots', async () => {
  const component = {
    components: {
      [Menu.name]: Menu,
      [MenuItem.name]: MenuItem,
    },
    template: `
      <m-menu>
        <m-menu-item name="1">
          <template #icon>
            <div>custom icon</div>
          </template>

          <template #label>
            <div>custom label</div>
          </template>
        </m-menu-item>
      </m-menu>
    `,
  }

  const wrapper = mount(component)
  expect(wrapper.html()).toMatchSnapshot()

  wrapper.destroy()
})
