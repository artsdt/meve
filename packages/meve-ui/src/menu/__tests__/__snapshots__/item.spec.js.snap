// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test menu item disabled 1`] = `
"<div class=\\"m-menu m--box\\">
  <div class=\\"m-menu-item m--box m-menu-item--disabled\\">
    <div class=\\"m-menu-item__indent\\"></div><i menu-item-cover=\\"true\\" class=\\"m-icon m--box meve-icon--set meve-icon-error m-menu-item__icon\\"></i>
    <div class=\\"m-menu-item__label\\">level 1</div>
  </div>
</div>"
`;

exports[`test menu item icon pass object 1`] = `
"<div class=\\"m-menu m--box\\">
  <div class=\\"m-menu-item m--box\\">
    <div class=\\"m-menu-item__indent\\"></div><i menu-item-cover=\\"true\\" class=\\"m-icon m--box meve-icon--set meve-icon-error m-menu-item__icon\\" style=\\"color: red; font-size: 30px;\\"></i>
    <div class=\\"m-menu-item__label\\">level i</div>
  </div>
</div>"
`;

exports[`test menu item pass render function 1`] = `
"<div class=\\"m-menu m--box\\">
  <div class=\\"m-menu-item m--box\\">
    <div class=\\"m-menu-item__indent\\"></div>
    <div>custom icon</div>
    <div>custom label</div>
  </div>
</div>"
`;

exports[`test menu item props 1`] = `
"<div class=\\"m-menu m--box\\">
  <div class=\\"m-menu-item m--box\\">
    <div class=\\"m-menu-item__indent\\"></div><i menu-item-cover=\\"true\\" class=\\"m-icon m--box meve-icon--set meve-icon-error m-menu-item__icon\\"></i>
    <div class=\\"m-menu-item__label\\">level 1</div>
  </div>
</div>"
`;

exports[`test menu item slots 1`] = `
"<div class=\\"m-menu m--box\\">
  <div class=\\"m-menu-item m--box\\">
    <div class=\\"m-menu-item__indent\\"></div>
    <div>custom icon</div>
    <div>custom label</div>
  </div>
</div>"
`;
